src/Adapter/CartRule/CommandHandler/AddCartRuleHandler.php

```diff
@@ -107,8 +109,15 @@ private function fillCartRuleConditionsFromCommandData(CartRule $cartRule, AddCa
     {
         $cartRule->id_customer = null !== $command->getCustomerId() ? $command->getCustomerId()->getValue() : null;
 
-        $cartRule->date_from = $command->getValidFrom()->format('Y-m-d H:i:s');
-        $cartRule->date_to = $command->getValidTo()->format('Y-m-d H:i:s');
+        if (null === $command->getValidFrom() || null === $command->getValidTo()) {
+            $now = new DateTimeImmutable();
+
+            $cartRule->date_from = $now->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT);
+            $cartRule->date_to = $now->modify('+1 month')->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT);
```

Comment:
This value (+1 month) has been decided with PMs ?

Reply:
No I haven't discussed this part particularly. I made it according to legacy behavior as the datepicker is by default prefilled with +1month value.